crates/router/src/services/api/client.rs

```diff
@@ -110,3 +116,145 @@ pub(super) fn proxy_bypass_urls(locker: &Locker) -> Vec<String> {
+    client: ProxyClient,
+}
+
+impl RequestBuilder for RouterRequestBuilder {
+    fn json(&mut self, body: serde_json::Value) {
+        self.inner = self.inner.take().map(|r| r.json(&body));
+    }
+    fn url_encoded_form(&mut self, body: serde_json::Value) {
+        self.inner = self.inner.take().map(|r| r.form(&body));
+    }
+
+    fn timeout(&mut self, timeout: Duration) {
+        self.inner = self.inner.take().map(|r| r.timeout(timeout));
+    }
+
+    fn multipart(&mut self, form: Form) {
+        self.inner = self.inner.take().map(|r| r.multipart(form));
+    }
+
+    fn header(&mut self, key: String, value: Maskable<String>) -> CustomResult<(), ApiClientError> {
+        let header_value = match value {
+            Maskable::Masked(hvalue) => {
+                HeaderValue::from_str(hvalue.peek()).map(|mut h| {
+                    h.set_sensitive(true);
+                    h
+                })
+            }
+            Maskable::Normal(hvalue) => HeaderValue::from_str(&hvalue),
+        }
+        .into_report()
+        .change_context(ApiClientError::HeaderMapConstructionFailed)?;
+
+        self.inner = self.inner.take().map(|r| r.header(key, header_value));
+        Ok(())
+    }
+
+    fn send(
+        self,
+    ) -> CustomResult<Box<(dyn core::future::Future<Output = Result<reqwest::Response, reqwest::Error>> + 'static)>, ApiClientError> {
+        Ok(Box::new(self.inner.ok_or(ApiClientError::UnexpectedState)?.send()))
+    }
+}
+
+// TODO: remove this when integrating this trait
+#[allow(dead_code)]
+impl ApiClient for ProxyClient {
+    fn new(
+        proxy_config: Proxy,
+        whitelisted_urls: Vec<String>,
+    ) -> CustomResult<Self, ApiClientError> {
+        let non_proxy_client = reqwest::Client::builder()
+            .redirect(reqwest::redirect::Policy::none())
+            .build()
+            .into_report().change_context(ApiClientError::ClientConstructionFailed)?;
+        
+        let mut proxy_builder =
+            reqwest::Client::builder().redirect(reqwest::redirect::Policy::none());
+
+        if let Some(url) = proxy_config.https_url.as_ref() {
+            proxy_builder = proxy_builder.proxy(reqwest::Proxy::https(url).into_report().change_context(ApiClientError::InvalidProxyConfiguration)?);
+        }
+
+        if let Some(url) = proxy_config.http_url.as_ref() {
+            proxy_builder = proxy_builder.proxy(reqwest::Proxy::http(url).into_report().change_context(ApiClientError::InvalidProxyConfiguration)?);
+        }
+
+        let proxy_client = proxy_builder.build().into_report().change_context(ApiClientError::InvalidProxyConfiguration)?; //todo
+        Ok(Self {
+            proxy_client,
+            non_proxy_client,
+            whitelisted_urls,
+        })
+    }
+
+    fn request<U: IntoUrl>(&self, method: Method, url: U) -> Box<dyn RequestBuilder> {
+        Box::new(RouterRequestBuilder {
+            inner: Some(self.proxy_client.request(method, url)),
```

Comment:
```suggestion
            inner: Some(self.get_reqwest_client(...)),
```

Also add the below  methods to `trait ApiClient` & use them in this function with the default values as None

```
pub fn request_with_certificate(&self, certificate: Option<String>, certificate_key: Option<String>);
```